datafusion/core/src/execution/context/mod.rs

```diff
@@ -859,7 +859,7 @@ impl SessionContext {
         // check if the file extension matches the expected extension
         for path in &table_paths {
             let file_name = path.prefix().filename().unwrap_or_default();
-            if !path.as_str().ends_with(&option_extension) && file_name.contains('.') {
+            if !file_name.ends_with(&option_extension) && file_name.contains('.') {
```

Comment:
I think this check is just ill-formed, if the ListingTableUrl is to a directory containing a `.` this check will reject it. I'm not sure what this check is here for tbh...